### ISSUE TITLE
profile is not updated when returning from editing dialog

### ISSUE THREAD

**Issue by niksauer** [niksauer/RoadChat#233]

**Comment by niksauer** [comment]
viewDidAppear is called up to FixedPageViewController

**Comment by grinchzema** [comment]
Hello. cloned from git RoadChat, but I do get frameworks empty. and this is the error that I get. Could you please help me figure it out ?
<img width="849" alt="Screen Shot 2019-08-21 at 12 24 31 PM" src="https://user-images.githubusercontent.com/21989487/63420161-d09c4800-c40e-11e9-8707-a4eccab2442e.png">

**Comment by niksauer** [comment]
Hi, have you updated the submodules afterwards?

`git submodule update --recursive --init`

**Comment by grinchzema** [comment]
hi, thank you for your reply.
the thing is I am using GitHub projects for the first time. on terminal I
did commands like git init.
and then git clone and the url of RoadChat.
then I went to all of submodules and downloaded them to my Mac.
thave never worked with a command like git submodule before. should just
type that right after git clone <url> finishes on the next line ?
I would be glad if you could guide me a bit

**Comment by niksauer** [comment]
Yeah no problem, so first of all you don't have to execute `git init` on this project. This has already been done at the very beginning of the project. Otherwise I wouldn't have been able to upload it to GitHub 😊 `git init` is only used when you want to put some directory under version control (aka Git). 

To set up this project:
```
git clone git@github.com:niksauer/RoadChat.git
cd RoadChat
git submodule update --recursive --init
```

Since you changed the directory to this Git project after cloning, it will know which submodules to download / update.

**Comment by grinchzema** [comment]
I tried the first command, this is what I get:

Cloning into 'RoadChat.git.'...

git@github.com: Permission denied (publickey).

fatal: Could not read from remote repository.


Please make sure you have the correct access rights

and the repository exists.

**Comment by niksauer** [comment]
Have a look [here](https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account). Otherwise you can use git via HTTPS: `git clone https://github.com/niksauer/RoadChat.git`

**Comment by grinchzema** [comment]
I am so sorry, I got stucked.
What I am doing wrong?

**Comment by niksauer** [comment]
Not sure what the problem is right now?

**Comment by grinchzema** [comment]
when I just say git clone https://github.com/niksauer/RoadChat.git
I says fatal: destination path 'RoadChat' already exists and is not an
empty directory
the thing I do not how to remove that path.
even with rm -r

**Comment by grinchzema** [comment]
it asks for user name and password when I enter it it gives an error:

Username for 'https://github.com': grinchzema@gmail.com

Password for 'https://grinchzema@gmail.com@github.com':

remote: Repository not found.

fatal: repository 'https://github.com/niksauer/RoadChat.git./' not found

Narmins-MacBook-Pro:~ narminmansimova$ git clone
https://github.com/niksauer/RoadChat.git

**Comment by niksauer** [comment]
`rm -rf RoadChat`

Please follow the guide to setup SSH keys. That will prevent a lot of problems in the future.

**Comment by grinchzema** [comment]
thank you that worked.
but after cloning submodules I get this for each of them

Please make sure you have the correct access rights

and the repository exists.

fatal: clone of 'git@github.com:rechsteiner/Parchment.git' into submodule
path '/Users/narminmansimova/RoadChat/Frameworks/Parchment' failed

Failed to clone 'Frameworks/Parchment' a second time, aborting

Cloning into '/Users/narminmansimova/RoadChat/Frameworks/NetworkKit'...

git@github.com: Permission denied (publickey).

fatal: Could not read from remote repository.

**Comment by niksauer** [comment]
Again, you will have to setup SSH keys according to https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account.

**Comment by grinchzema** [comment]
hi,
thank you so much.
finally the build succeeded. :)
this whenever I press Register or login nothing happens.
I pressed several times.
and this is what I got on console.

*[WC] -[WCSession onqueue_handleUpdateSessionState:]_block_invoke dropping
as pairingIDs no longer match. pairingID (null), client pairingID: (null)*

*2019-08-21 16:21:39.646150+0300 RoadChat[28275:320409] [WC] WCSession is
not paired*

*2019-08-21 16:21:51.455979+0300 RoadChat[28275:320408] [AutoFill] Cannot
show Automatic Strong Passwords for app bundleID: com.hpe.dhbw.RoadChat due
to error: iCloud Keychain is disabled*

**Comment by grinchzema** [comment]
should I create a new database ?
I swear this is the last issue , so sorry for disturbing you

**Comment by niksauer** [comment]
No worries, happy to see someone learning from my projects. 

You'll have to be running the [matching backend](https://github.com/niksauer/RoadChatAPI) either locally or on a remote server of your choice. The url can than be configured [here](https://github.com/niksauer/RoadChat/blob/master/RoadChat/Frontend/Model/RoadChatAPI.swift).

**Comment by grinchzema** [comment]
could you please look at my 1 previous message just before the database one
?
there is an issue while I hit register button after filling all of the
fields.
nothing happens at all.
it does not direct me to the app. I just stay on register scene

**Comment by niksauer** [comment]
In order to register, the app has to communicate with the RoadChatAPI backend. Therefore you must download and run that project as well.

**Comment by grinchzema** [comment]
I guess this is what you are meaning right?
I do already have it inside. it builds successfully.
except for it stays on that page.
could you please guide a bit more into communication with the API backend.
the only backend I have worked so far is Firebase . So I do not have prev
knowledge .
Thank you for your support

**Comment by grinchzema** [comment]
I just do not know how to get a local/remote host.
I googled before asking for your assistance

**Comment by grinchzema** [comment]
I downloaded RoadChatAPI into frameworks, still confused.
I am sorry again for disturbing.
Have a nice day

**Comment by niksauer** [comment]
It’s not a framework. You have to clone the project seperately and follow the README, i.e set up the Swift Package Manager, Docker etc. I can‘t help you with all the steps, it‘s kinda advanced.
